internal/sshtunneler/tunneler.go

```diff
@@ -0,0 +1,278 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshtunneler
+
+import (
+	"context"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/juju/clock"
+	"github.com/juju/errors"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	gossh "golang.org/x/crypto/ssh"
+
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/pki/ssh"
+	"github.com/juju/juju/state"
+)
+
+const (
+	maxTimeout        = 60 * time.Second
+	reverseTunnelUser = "reverse-tunnel"
```

Comment:
Should be something like `juju-reverse-tunnel` and we might need to ensure no one can create a user with this username if the plan is that the ssh proxy server is terminating both inbound connections from JIMM/End-Users and from machine/unit agents. If this rendezvous work here is using a separate listener then this is of no concern.